src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -468,6 +464,27 @@ private bool TryGetWorkspaceFixersPriorityMap(Document document, [NotNullWhen(tr
             }
         }
 
+        private CodeChangeProviderMetadata? TryGetMetadata(CodeFixProvider fixer)
+        {
+            return ImmutableInterlocked.GetOrAdd(
+                ref _fixerToMetadataMap,
+                fixer,
+                static (fixer, fixers) =>
+                {
+                    foreach (var lazy in fixers)
+                    {
+                        if (lazy.IsValueCreated && lazy.Value == fixer)
+                            return lazy.Metadata;
+                    }
+
+                    // Note: it feels very strange that we could ever not find a fixer in our list.  However, this
+                    // occurs in testing scenarios.  I'm not sure if the tests represent a bogus potential input, or if
+                    // this is something that can actually occur in practice and we want to keep working.
+                    return null;
```

Comment:
@mavasani for thoughts on this.   Specifically, if i throw here, tests in CodeFixService tests fail.  but i think they're likely constructing this service wrong.

Reply:
for example it does:

```c#
            var fixService = new CodeFixService(
                diagnosticService, logger, fixers, SpecializedCollections.EmptyEnumerable<Lazy<IConfigurationFixProvider, CodeChangeProviderMetadata>>());
```

So we have a list of fixers, but no metadata about htem.  i can't tell if htis is a reasonable or rational think that coudl actually happen in practice.

Reply:
Do you recall which tests were failing? From what I vaguely recall, we had some scenario where if a code fixer throws some exception during creation due to some missing types or something, we wouldn't be able to create an instance of the fixer, and had to handle this error case with nulls. I am wondering if the failures you saw were coming from tests explicitly validating exception during fixer creation.

Reply:
yeah, it was tests in CodeFixServiceTets.  THere we manually instnatiated some test-only CodeFixProviders, then instantiated the CodeFixService.  But we passed along an empty metadata-array, presumably because we didn't care in those tests and the code was resilient to not having it.  But i can't tell if that's a real scenario taht occurs in practice.